### PR TITLE
Add ADR for My Facilities dashboard

### PR DESCRIPTION
**Pull request by prabhanshuguptagit**
https://www.pivotaltracker.com/story/show/170814678

### CODE REVIEW

File: doc/arch/008-materialized-views-for-dashboards.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+Materialized views are a good intermediate solution, since they give us many of the benefits of an ETL pipeline,
+without the overhead of specialised infrastructure. We may need to review this decision if materialized view refreshes become very expensive.
+We will use [scenic](https://github.com/scenic-views/scenic) to manage the views.
+
+## Approach
+### Defining the views
+* For each dashboard, we may define a view to represent its data, or choose to query existing views.
+* The views will slice data by the time periods we need to query by (eg: `days`, `months`, `quarters`).
+* Ideally, the view definitions should be free of domain logic.
```
Comment (harimohanraj89) [inline doc/arch/008-materialized-views-for-dashboards.md]: Can you include an example like `hypertensive` here?

File: doc/arch/008-materialized-views-for-dashboards.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+### Using the views
+* Views will have to be refreshed periodically. The period can vary depending on the dashboard's requirement.
+* Each dashboard will consist of a query object, the necessary materialized views for its data, and templates to render the data appropriately.
```
Comment (harimohanraj89) [inline doc/arch/008-materialized-views-for-dashboards.md]: `PORO query object`? Can we link to relevant lines of code as examples? We can freeze the link to the current commit so this documentation doesn't go out of date as the codebase evolves.

File: doc/arch/008-materialized-views-for-dashboards.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+* The query object will provide an interface to fetch the data from the views. These objects should return active record relations to allow for composability and reuse.
+ 
+## Status
+
+Accepted
+
+## Consequences
+
+* Materialized view refresh times will scale with the size of the database tables. Currently `blood_pressures` is the most significant table.
```
Comment (harimohanraj89) [inline doc/arch/008-materialized-views-for-dashboards.md]: Do we have measurements we can quote here? Eg. "the refresh takes xxxxx seconds when the BP table has yyyyyyyy records"?

File: doc/arch/008-materialized-views-for-dashboards.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+* View refresh order is important, dependant views should be refreshed only after their parent views to ensure consistency.
```
Comment (harimohanraj89) [inline doc/arch/008-materialized-views-for-dashboards.md]: Can we link to the rake/clock task that was created to ensure the views are refreshed in the correct order?

Comment (harimohanraj89) [reply doc/arch/008-materialized-views-for-dashboards.md]: PS, have we kept that task up-to-date as we added new views?

File: doc/arch/008-materialized-views-for-dashboards.md (hunk 2 above)
Comment (vkrmis) [reply doc/arch/008-materialized-views-for-dashboards.md]: Yup, adding a link

File: doc/arch/008-materialized-views-for-dashboards.md (hunk 4 above)
Comment (vkrmis) [reply doc/arch/008-materialized-views-for-dashboards.md]: Added the link, and yes, the task is up-to-date.

File: doc/arch/008-materialized-views-for-dashboards.md (hunk 3 above)
Comment (vkrmis) [reply doc/arch/008-materialized-views-for-dashboards.md]: @prabhanshuguptagit can you add this? I don't have a large data set locally.

Comment (harimohanraj89) [reply doc/arch/008-materialized-views-for-dashboards.md]: _If_ we knew, we should add to this doc. If not, don't sweat it.

Comment (prabhanshuguptagit) [reply doc/arch/008-materialized-views-for-dashboards.md]: Added

**Pull request merged**